src/components/dropdownSearch/DropdownSearchReducers.tsx

```diff
@@ -44,9 +50,21 @@ export const getNextIndexPosition = (array: any[], item: any, key: number): numb
   return index;
 };
 
-export const getOptionsFiltered = (state: IDropdownSearchState, filterText?: string) => {
-  const currentFilterText: string = filterText || state.filterText;
+export const removeSelectedOption = (state: IDropdownSearchState, displayValue: string): FixedQueue<IDropdownOption> =>
+  state.selectedOptions.removeElementsWithProperties({ displayValue });
+
+export const addUniqueSelectedOption = (state: IDropdownSearchState, displayValue: string): FixedQueue<IDropdownOption> =>
```

Comment:
Unique means that the option you add will be trimmed out if it already exists. It removes any instances of the same displayValue in the selectedOptions and then add the new one